client/src/components/WeatherNav.vue

```diff
@@ -44,22 +56,106 @@
   export default {
     name: 'WeatherNav',
     components: {NavItem},
-    props: ['sitestatus'],
     data: function(){
       return {
         sites: sites,
-        timeRange: '24:hours'
+        timeRange: '24:hours',
+        site_statuses: {}
       };
     },
-    computed:{
-      num(){
+    methods:
+      {
+        fetch_site_status(site_code, start, end, callback)
+        /**
+         * Returns a letter (Y or N or ?) corresponding to the site status.
+         * @param site_code : The 3-letter side code.
+         * @param start: Query start, determined by start property
+         * @param end: query end, determined by end property
+         * @param callback: Handles the response received from Weather API
+         */
+        {
+          let request = new XMLHttpRequest();
+          let url = 'https://weather-api.lco.global/query?site=' + site_code + '&datumname=' + 'Weather Ok To Open' +
+            '&agg=False' + '&start=' + start.format() + '&end=' + end.format();
+          request.open('GET', url, false);
+          request.onload = () => {
+            if (request.status >= 200 && request.status < 400)
+            {
+              callback(JSON.parse(request.responseText));
+
+            }
+            else
+            {
+              console.log("error, couldn't receive response");
+            }
+          };
+
+          request.onerror = function()
+          {
+            console.log("A connection error occurred.");
+          };
+
+          request.send();
+        },
+
+        status(site_code)
+        {
+          /**
+           * Calls the internal fetch_site_status function.
+           */
+          var my_status;
+          this.fetch_site_status(site_code, this.start, this.$store.getters.end, (resp) =>
+          {
+            if (!resp || resp.length < 1)
+            {
+              this.site_statuses[site_code] = '?';
+              my_status = '?';
+            }
+            else {
+              var last_val = resp[resp.length - 1].ValueString;
+              var last_letter = this.$options.filters.statusMsgToLetter(last_val);
```

Comment:
Filters are only meant to be used in templates. If this is just a helper method, no need to place it in filters. Use a computed or normal method instead.

Reply:
Okay, this code was moved to the methods() section.